query-service-impl/src/main/java/org/hypertrace/core/query/service/projection/ProjectionTransformation.java

```diff
@@ -323,17 +324,17 @@ private QueryRequest rebuildRequest(
       QueryRequest original,
       List<Expression> selections,
       List<Expression> aggregations,
-      Filter filter,
+      Filter originalFilter,
       List<Expression> groupBys,
       List<OrderByExpression> orderBys) {
 
     QueryRequest.Builder builder = original.toBuilder();
-    filter = updateFilterForComplexAttributeExpression(filter, orderBys);
+    Filter updatedFilter = rebuildFilterForComplexAttributeExpression(originalFilter, orderBys);
 
-    if (Filter.getDefaultInstance().equals(filter)) {
+    if (Filter.getDefaultInstance().equals(updatedFilter)) {
```

Comment:
I think eventually this check would be required as the final filter can always be empty. So either we pass the builder to perform the check there or keep it as it is.

Reply:
The final filter can't be empty if the original one wasn't (otherwise we're removing a requested filter) - that's why I moved the check in the snippet above to before we run the rebuild function.

Reply:
You are performing a check with updatedFilter which is set from that function only.

Reply:
Sorry, copy pasted because I'm typing in the browser.

```diff
+ if (!Filter.getDefaultInstance().equals(originalFilter)) {
- if (!Filter.getDefaultInstance().equals(updatedFilter)) {
```

Reply:
What do we do in else block ?
There could be a case that the filter is empty but we need to add a filter from the order by right ?

Reply:
You're right, my bad. In practice it won't happen, there will never be a query without a filter since we'll always have a tenant and time filter, but accounting for the case is the right thing to do. So if the original filter isn't empty, the final filter won't be, but if the original filter is empty, we may still add a filter here. So we'd always run it, and change the condition to remove the clear case only.

```java
QueryRequest.Builder builder = original.toBuilder();
Filter updatedFilter = rebuildFilterForComplexAttributeExpression(originalFilter, orderBys);
if (!Filter.getDefaultInstance().equals(updatedFilter)) { 
  builder.setFilter(updatedFilter);
}
```

Reply:
resolved